[WIP] Do not use VERSION and VERSIONS for the same purpose in common.mk

pkubatrh opened the pull request:
Instead only use VERSIONS to choose which versions of the image to build in the makefile
and depend on its rules for populating the VERSION variable to be used in build.sh

Based on https://github.com/sclorg/postgresql-container/pull/152

omron93 commented:
@bparees What backward compatibility we have to keep in build scripts?

praiskup commented on common.mk:
```diff
@@ -17,25 +17,24 @@ endif
 .PHONY: build
-build: manpages
-	$(script_env) $(build)
+build: $(VERSIONS)
```
This will cause parallel build of multiple image versions, right?  What's the status of docker bug discussed in  sclorg/postgresql-container/pull/149 ?

pkubatrh replied:
Correct, if the user has setup to run make in parallel by default. But I agree this is still an issue. 

The bug mentioned in  sclorg/postgresql-container#149 happens when testing, not building, so to work around this issue for the time being we could leave the build rule as-is and modify the test rule to not be able to be run in parallel (possibly by separating `build.sh` into building and testing parts)

pkubatrh commented:
Pushed a new version of the changes, this time making sure test suites for different versions of the image are not able to be run in parallel by splitting off the testing code located in `build.sh` into `test.sh`.

That said I have once again tried running the test suites in parallel (on a rhel-7.3 machine) using code from sclorg/postgresql-container#149 (without any of the "hack" commits) and have yet to hit a failure over ~200 runs.

Eventually it would make sense to move most, if not all, of `test.sh` code into the `common.mk`.

omron93 commented:
Not only tests of postgresql-container are not able to run in parallel. Also a lot of s2i based images runs `s2i build` in tests and name of resulted image does not differ among versions.

So they have to be fixed too before enabling parallel testing.

praiskup commented:
Sorry, I totally forgot about this PR;  and the PR #12 caused collisions :(.  Can this be rebased please?

pkubatrh commented:
Rebased. Additionally I split off the tagging code to a separate script as well. I am wondering if it makes sense to introduce a `tag` rule to the Makefile or if it's ok to leave it as is.

@praiskup @omron93 PTAL

omron93 commented:
It looks really good. Thanks for a lot of improvements.

+1 for merging

praiskup commented:
Agreed, this is very nice and clean first step!  Thanks.
